Port MASTG-TEST-0022: Testing Custom Certificate Stores and Certificate Pinning (android) (by @guardsquare) (#3035)
OWASP/owasp-mastg: tests-beta/android/MASVS-NETWORK/MASTG-TEST-0x22-2.md

---
title: Expired Certificate Pins
platform: android

> id: MASTG-TEST-0x22-2

Reviewer: Could you please update each test and give it a unique ID like here: https://github.com/OWASP/owasp-mastg/tree/master/tests-beta/android/MASVS-NETWORK. This means also changing the filename accordingly.

type: [static]
weakness: MASWE-0047
---

## Overview

> Apps can configure expiration dates for pinned certificates in the [Network Security Configuration]("../../../Document/0x05g-Testing-Network-Communication.md#certificate-pinning"). After the expiration date the pin is not used any more, and all installed CAs are trusted for that domain.

Reviewer: ```suggestion
Apps can configure expiration dates for pinned certificates in the [Network Security Configuration]("../../../Document/0x05g-Testing-Network-Communication.md#certificate-pinning"). After the expiration date, the pin is no longer used and all installed CAs for that domain are trusted.
```
